test/it/RetrieveLegalUnitBYKeyAcceptanceSpec.scala

```diff
@@ -0,0 +1,94 @@
+      stubHBaseFor(aLegalUnitRequest(withErn = TargetErn, withPeriod = TargetPeriod, withUBRN = TargetUBRN).willReturn(
+        anOkResponse().withBody(LegalUnitSingleMatchHBaseResponseBody)
+      ))
+
+      When(s"the legal unit with $TargetErn, $TargetPeriod, and $TargetUBRN is requested")
+      val response = await(wsClient.url(s"/v1/enterprises/${TargetErn.value}/periods/${Period.asString(TargetPeriod)}/legalunits/${TargetUBRN.value}").get())
+
+      Then(s"the details of the unique legal unit identified by $TargetErn, $TargetPeriod, and $TargetUBRN are returned")
+      response.status shouldBe OK
+      response.header(CONTENT_TYPE).value shouldBe JSON
+      response.json.as[LegalUnit] shouldBe
+        LegalUnit(TargetUBRN, UBRNref = Some("some-UBRN"), enterprise = EnterpriseLink(TargetErn, entref = Some("some-entref")))
+    }
+  }
+  feature("retrieve a non-existent Legal Unit") {
+    scenario(s"by exact Enterprise reference (ERN), period, and Legal Unit reference (UBRN)") { wsClient =>
+      Given(s"a legal unit does not exist with $TargetErn, $TargetPeriod, and $TargetUBRN")
+      stubHBaseFor(aLegalUnitRequest(withErn = TargetErn, withPeriod = TargetPeriod, withUBRN = TargetUBRN).willReturn(
+        anOkResponse().withBody(NoMatchFoundResponse)
+      ))
+
+      When(s"a legal unit with $TargetErn, $TargetPeriod, and $TargetUBRN is requested")
+      val response = await(wsClient.url(s"/v1/enterprises/${TargetErn.value}/periods/${Period.asString(TargetPeriod)}/legalunits/${TargetUBRN.value}").get())
+
+      Then(s"a NOT FOUND response is returned")
+      response.status shouldBe NOT_FOUND
+    }
+  }
+  feature("validate request parameters") {
+    scenario(s"rejecting an Enterprise reference (ERN) that is not ten digits long") { wsClient =>
+      Given(s"that an ERN is represented by a ten digit number")
+
+      When(s"the user requests a Leqal Unit having an ERN that is not ten digits long")
+      val response = await(wsClient.url(s"/v1/enterprises/123456789/periods/${Period.asString(TargetPeriod)}/legalunits/${TargetUBRN.value}").get())
+
+      Then(s"a BAD REQUEST response is returned")
+      response.status shouldBe BAD_REQUEST
+    }
+  }
+  feature("handle inability to connect to HBase REST service") {
+    scenario("WireMock stops, imitating a loss of connection to HBase") { wsClient =>
+      Given("request sent to the HBase REST mock")
+      stubHBaseFor(aLegalUnitRequest(withErn = TargetErn, withPeriod = TargetPeriod, withUBRN = TargetUBRN).willReturn(
+        anOkResponse().withBody(LegalUnitSingleMatchHBaseResponseBody)
+      ))
+
+      When("the mock of HBase stops")
+      stopWireMock()
+      val response = await(wsClient.url(s"/v1/enterprises/${TargetErn.value}/periods/${Period.asString(TargetPeriod)}/legalunits/${TargetUBRN.value}").get())
+
+      Then("an INTERNAL SERVER ERROR response is returned")
+      response.status shouldBe INTERNAL_SERVER_ERROR
+    }
```

Comment:
This doesn't really tell the story from the user perspective.  Aim for something like:

Given("the legal unit repository is unavailable")
stopWireMock()

When("the user requests a legal unit")
val response = await(wsClient.url(...

Then("an INTERNAL SERVER ERROR response is returned")
response.status shouldBe INTERNAL_SERVER_ERROR